src/app/templates/ExpensesView.tsx

```diff
@@ -64,56 +71,149 @@ const ExpensesView: FC<ExpensesViewProps> = ({
+      const defaultGasFee = mutezToTz(defaultGasFeeMutez);
       const storageFee = mutezToTz(storageFeeMutez);
 
       return (
         <div className="w-full flex flex-col">
           {[
-            { key: "gas", title: t("gasFee"), fee: gasFee },
             {
-              key: "storage",
+              key: "totalFee",
+              title: t("gasFee"),
+              value: gasFee,
+              onChange: modifyFeeAndLimit.onTotalFeeChange,
+            },
+            {
+              key: "storageFeeMax",
               title: t("storageFeeMax"),
-              fee: storageFee,
+              value: storageFee,
             },
-          ].map(({ key, title, fee }) => (
-            <div key={key} className="mb-px w-full flex items-center">
+            ...(modifyFeeAndLimit.storageLimit !== null
+              ? [
+                  {
+                    key: "storageLimit",
+                    title: t("storageLimit"),
+                    value: modifyFeeAndLimit.storageLimit,
+                    onChange: modifyFeeAndLimit.onStorageLimitChange,
+                  },
+                ]
+              : []),
+          ].map(({ key, title, value, onChange }, i, arr) => (
+            <div
+              key={key}
+              className={classNames(
+                "w-full flex items-center",
+                i !== arr.length - 1 && "mb-1"
+              )}
+            >
               <div
                 className={classNames(
-                  "mr-1",
                   "whitespace-no-wrap overflow-x-auto no-scrollbar",
                   "opacity-90"
                 )}
-                style={{ maxWidth: "40%" }}
+                style={{ maxWidth: "45%" }}
               >
-                {title}:
+                {title}
               </div>
+              <div className="mr-1">:</div>
+
               <div className="flex-1" />
-              <div className="font-medium mr-1">
-                <Money>{fee}</Money> ꜩ
-              </div>
-              <InUSD
-                volume={fee}
-                roundingMode={BigNumber.ROUND_UP}
-                mainnet={mainnet}
-              >
-                {(usdAmount) => (
-                  <div>
-                    <span className="opacity-75">(</span>
-                    <span className="pr-px">$</span>
-                    {usdAmount}
-                    <span className="opacity-75">)</span>
+
+              {value instanceof BigNumber ? (
+                <>
+                  <div className="mr-1">
+                    {onChange ? (
+                      <>
+                        <PlainAssetInput
+                          value={value.toFixed()}
+                          onChange={(val) => {
+                            onChange?.(
+                              tzToMutez(val ?? defaultGasFee).toNumber()
+                            );
+                          }}
+                          min={0}
```

Comment:
I'm afraid we need to set a minimal value of 1e-6 or, maybe, something like `defaultGasFee * 0.9` because it's now possible to create a transaction with zero gas fee and it even doesn't go to mempool for a long time
![Screenshot_2021-06-15_09-03-51](https://user-images.githubusercontent.com/10885399/122001787-82adbc80-cdb9-11eb-88fb-4746a3b51b79.png)
![Screenshot_2021-06-15_09-09-30](https://user-images.githubusercontent.com/10885399/122001791-83dee980-cdb9-11eb-9015-1258391a0236.png)

Reply:
![Screenshot_2021-06-15_09-11-54](https://user-images.githubusercontent.com/10885399/122002090-f8b22380-cdb9-11eb-9002-de1344fe0376.png)
![Screenshot_2021-06-15_09-14-08](https://user-images.githubusercontent.com/10885399/122002175-14b5c500-cdba-11eb-8611-5e7bdd36cfd3.png)
Oy vey, it looks like we need a minimal value which is equal to `defaultGasFee`

Reply:
Here is the same transaction with minimal gas fee, and it is completed: https://tzkt.io/opJYyrg2ia1UHfec3isGKGdUGC9jXYonhZVdfdpNX8dqbznWxXs
It's just for reference